aim2dat/strct/structure_operations.py

```diff
@@ -44,7 +47,13 @@ def _create_index_combinations(confined, strct_c_len):
     pairs = []
     for idx0 in range(strct_c_len):
         for idx1 in range(min_idx, max_idx):
-            if idx0 != idx1 and (idx1, idx0) not in pairs:
+            if (
+                utils_cf.compare_formulas(
+                    strct_c[idx0].chem_formula, strct_c[idx1].chem_formula, reduce_formulas=True
+                )
+                and idx0 != idx1
+                and (idx1, idx0) not in pairs
+            ):
```

Comment:
```suggestion
            if (
                idx0 != idx1
                and (idx1, idx0) not in pairs
                and utils_cf.compare_formulas(
                    strct_c[idx0].chem_formula, strct_c[idx1].chem_formula, reduce_formulas=True
                )
            ):
```

The improvement in terms of speed will be small since the comparison of the formulas is quite cheap. Nonetheless, the comparison will be skipped if the indices are identical or the pair already exists.